ceremonies/src/lib.rs

```diff
@@ -138,14 +157,16 @@ decl_module! {
 
             let cindex = <encointer_scheduler::Module<T>>::current_ceremony_index();
 
-            if <ParticipantIndex<T>>::contains_key((cid, cindex), &sender) {
+            if <BootstrapperIndex<T>>::contains_key((cid, cindex), &sender) ||
+                <ReputableIndex<T>>::contains_key((cid, cindex), &sender)||
+                <EndorseeIndex<T>>::contains_key((cid, cindex), &sender)||
+                <NewbieIndex<T>>::contains_key((cid, cindex), &sender)
```

Comment:
this checking of presence of the sender in any of the indices probably corresponds to a check of existence or similar.. one could extract this into a method with parameters (cid, cindex, sender) and name the question accordingly..

when changing the data structure (unprobable on blockchain, i know), the phrasing of the method can provide hints of  whether to change the logic in it or not..